tensorflow/core/kernels/mkl/mkl_matmul_ops_common.h

```diff
@@ -487,6 +487,54 @@ class MklDnnMatMulOpBase : public OpKernel {
     return (weight_oi_.NumElements() == 0);
   }
 
+  // Cache the converted weight in a tensor.
+  // Only one thread can execute this method at any given time.
+  void CacheWeight(
+      OpKernelContext* context,
+      const std::shared_ptr<dnnl::matmul::primitive_desc>&
+          matmul_fwd_pd,
+      Tweight* weight_data, const Tensor& weight_tensor,
+      MklDnnData<Tweight>& weight, const memory::desc& weight_md)
+      TF_LOCKS_EXCLUDED(mu_) {
```

Comment:
This function is exactly the same as the existing `CacheWeight` for `dnnl::inner_product_forward`. Can we just make the old `CacheWeight` a template function instead? E.g,
```c++
  template <typename T>
  void CacheWeight(
      OpKernelContext* context,
      const std::shared_ptr<T>& matmul_fwd_pd,
      ...
```

Reply:
Thanks for the review @penpornk. I overloaded the function to customize it, but ended up keeping the same logic as the original, will make it templated.